test/unit/StakingProducts/setProducts.js

```diff
@@ -444,4 +444,23 @@ describe('setProducts unit tests', function () {
       initialProducts.map(p => p.productId),
     );
   });
+
+  it('should fail to recalculate effective weight for product not in this pool', async function () {
```

Comment:
TODO: change test to check that weight doesn't get effected by non pool products